fission/src/systems/scene/SceneRenderer.ts

```diff
@@ -166,6 +158,75 @@ class SceneRenderer extends WorldSystem {
         this.RemoveAllSceneObjects()
     }
 
+    /**
+     * Changes the quality of lighting between cascading shadows and directional lights
+     *
+     * @param quality: string representing the quality of lighting - "Low", "Medium", "High"
+     */
+    public ChangeLighting(quality: string): void {
+        // removing the previous lighting method
+        if (this._light instanceof THREE.DirectionalLight) {
+            this._scene.remove(this._light)
+        } else if (this._light instanceof CSM) {
+            this._light.dispose()
+            this._light.remove()
+        }
+
+        // setting the shadow map size
+        const shadowMapSize = Math.min(4096, this._renderer.capabilities.maxTextureSize)
+
+        // setting the light to a basic directional light
+        if (quality === "Low" || quality === "Medium") {
+            const shadowCamSize = 15
+
+            this._light = new THREE.DirectionalLight(0xffffff, 5.0)
+            this._light.position.set(-1.0, 3.0, 2.0)
+            this._light.castShadow = true
+            this._light.shadow.camera.top = shadowCamSize
+            this._light.shadow.camera.bottom = -shadowCamSize
+            this._light.shadow.camera.left = -shadowCamSize
+            this._light.shadow.camera.right = shadowCamSize
+            this._light.shadow.mapSize = new THREE.Vector2(shadowMapSize, shadowMapSize)
+            this._light.shadow.blurSamples = 16
+            this._light.shadow.bias = 0.0
+            this._scene.add(this._light)
+        }
+
+        // setting light to cascading shadows
+        else if (quality === "High") {
+            this._light = new CSM({
+                parent: this._scene,
+                camera: this._mainCamera,
+                cascades: 3,
+                lightDirection: new THREE.Vector3(0.5, -0.5, 0.5).normalize(),
+                lightIntensity: 5,
+                shadowMapSize: shadowMapSize,
+                mode: "custom",
+                maxFar: 30,
```

Comment:
Adding a shadow bias of `-0.00001` worked pretty well for the closest cascade, but not so much for the others. I'd tinker with it.